samples/19.a.single-sign-on-for-enterprise-apps/README.md

```diff
@@ -0,0 +1,195 @@
+
+Instead of OpenID, most enterprise apps use OAuth plus a graph API to identify an individual user. In this demo, we will demonstrate how to use OAuth to obtain access to graph API and use the API to identifying the accessor.
+
+This demo does not include any threat models and is designed for educational purpose only. When you design a production system, threat-modelling is an important task to make sure your system is secure and provide a way to quickly identify potential source of data breaches. IETF [RFC 6819](https://tools.ietf.org/html/rfc6819) is a good starting point for threat-modelling when using OAuth 2.0.
+
+# Assumptions
+
+- Developer have an existing enterprise web app that use OAuth to access protected resources
+   - We assume the OAuth access token lives in the browser memory and is accessible using JavaScript
+      - Access token can live in browser memory but must be secured during transmit thru the use of TLS
+      - More about security considerations can be found at [IETF RFC 6749 Section 10.3](https://tools.ietf.org/html/rfc6749#section-10.3)
+- Developer know how to alter existing JavaScript code around their existing UI for OAuth
+
+# Goals
+
+- Website supports both anonymous and authenticated access. Forced page refresh and/or new conversation is not mandated
+- End-user is able to sign in thru the web page, and recognized by the bot immediately
+- End-user is able to sign in thru the bot, and recognized by the web page immediately
+- End-user is able to sign in thru the web page, and sign out thru the bot
+- End-user is able to sign in thru the bot, and sign out thru the web page
+- OAuth access token is saved as part of the conversation state or user state
+
+# How to run this demo
+
+> For quickstart, you can browse to https://webchat-sso.azurewebsites.net/ to try out this demo in our hosted environment.
+
+This demo integrates with multiple services. There are multiple services you need to setup in order to host the demo.
+
+1. [Clone the code](#clone-the-code)
+1. [Setup OAuth via Azure Active Directory](#setup-oauth-via-azure-active-directory)
+1. [Setup OAuth via GitHub](#setup-oauth-via-github)
+1. [Setup Azure Bot Services](#setup-azure-bot-services)
+1. [Prepare and run the code](#prepare-and-run-the-code)
+
+## Clone the code
+
+To host this demo, you will need to clone the code and run locally.
+
+1. Clone this repository
+1. Create two files for environment variables, `/bot/.env` and `/rest-api/.env`
+   - In `/rest-api/.env`
+      - Write `AAD_OAUTH_REDIRECT_URI=http://localhost:3000/api/aad/oauth/callback`
+         - When Azure Active Directory complete the authorization flow, it will send the browser to this URL. This URL must be accessible by the browser from end-user machine
+      - Write `GITHUB_OAUTH_REDIRECT_URI=http://localhost:3000/api/github/oauth/callback`
+         - Same as Azure Active Directory, this is the URL for GitHub to send its result
+
+## Setup OAuth via Azure Active Directory
```

Comment:
Move this section `## Setup OAuth via Azure Active Directory` to below `Setup OAuth via GitHub`

Reply:
Reordered.